packages/suite/src/components/suite/OverflowItems.tsx

```diff
@@ -0,0 +1,130 @@
+import { ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+import { usePrevious, useUpdateEffect } from 'react-use';
+import { useSelector } from 'src/hooks/suite';
+import styled from 'styled-components';
+
+export type OverflowItem = {
+    removeOrder?: number;
+    [key: string]: any;
+};
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-wrap: nowrap;
+    min-width: 0;
+`;
+
+interface OverflowItemsProps<T> {
+    className?: string;
+    items: T[];
+    visibleItemRenderer: (item: T, index: number) => ReactNode;
+    overflowRenderer: (items: T[]) => ReactNode;
+    minVisibleItems?: number;
+    alwaysRenderOverflow?: boolean;
+}
+
+export function OverflowItems<T extends OverflowItem>({
+    items,
+    className,
+    minVisibleItems = 0,
+    alwaysRenderOverflow = false,
+    visibleItemRenderer,
+    overflowRenderer,
+}: OverflowItemsProps<T>) {
+    const [state, setState] = useState({
+        visibleItems: items,
+        overflowItems: [] as T[],
+    });
+
+    const wrapperRef = useRef(null);
+
+    const showAllItems = useCallback(
+        () =>
+            setState({
+                overflowItems: [],
+                visibleItems: items,
+            }),
+        [items],
+    );
+
+    const repartition = useCallback(() => {
+        if (!wrapperRef.current) {
+            return;
+        }
+
+        const wrapper = wrapperRef.current as Element;
+
+        if (!wrapper) {
+            return;
+        }
+
+        if (
+            Math.floor(wrapper.scrollWidth - 1) > Math.floor(wrapper.getBoundingClientRect().width)
+        ) {
+            const getNewState = (prevState: typeof state) => {
+                if (prevState.visibleItems.length <= minVisibleItems!) {
+                    return prevState;
+                }
+
+                const nextItem = prevState.visibleItems
+                    .slice()
+                    .sort((item1, item2) => (item1.removeOrder ?? 0) - (item2.removeOrder ?? 0))[0];
+
+                if (!nextItem) {
+                    return prevState;
+                }
+
+                const visible = prevState.visibleItems.slice();
+                visible.splice(
+                    prevState.visibleItems.findIndex(
+                        item => item.removeOrder === nextItem.removeOrder,
+                    ),
+                    1,
+                );
+
+                return {
+                    overflowItems: [nextItem, ...prevState.overflowItems],
+                    visibleItems: visible,
+                };
+            };
+
+            setState(getNewState);
+        }
+    }, [minVisibleItems]);
+
+    const screenWidth = useSelector(state => state.resize.screenWidth) ?? 0;
```

Comment:
would be nice to keep component state and selectors above callbacks and effects unless not possible. It helps readability. It's not strictly enforced but everyone more or less is following [the same order](https://www.notion.so/satoshilabs/Components-9159b639d21b43ef8b66be150854b6b6?pvs=4#308aa7db519e4f4eaf70312f60ce030a).

Reply:
https://github.com/trezor/trezor-suite/pull/9722/commits/9dacbbb1e944ac9b4042d3097509aff4a2207a08 fixed